hipcheck/src/plugin/manager.rs

```
@@ -65,6 +168,39 @@ impl PluginExecutor {
 		Err(hc_error!("Failed to find available port"))
 	}
 
+	/// Launches async forwarding of plugin logs to program output.
+	fn forward_plugin_logs(
+		&self,
+		reader: impl tokio::io::AsyncRead + Unpin + Send + 'static,
+		plugin_title: String,
+	) {
+		tokio::spawn(async move {
+			let reader = BufReader::new(reader);
+			let mut lines = reader.lines();
+			// parse plugin log json lines to LogEvent objects and output the messages
+			while let Ok(Some(line)) = lines.next_line().await {
```

As I understand it, this will silently ignore the `Err()` case, which would contain a `std::io::Error` if there is an I/O error when reading from the plugin's log stream.

While we probably can't do much to respond to such an error, I wonder if it's valuable to report it by logging? It could be a useful hint that something is going wrong, though I've seen quite a few cases where it's hard to avoid I/O errors during shutdown of IPC connections. Issues would likely be platform-dependent too.

---

I don't think it hurts to follow Cal's suggestion of notifying the user when I/O with the child process has failed before this async task dies. e.g.

```rust
loop {
    let line = match lines.next_line().await {
       Ok(Some(l)) => l,
       Ok(None) => {break; },
       Err(e) => {
             log::error!("I/O erorr occured while listening for logs from {plugin_title}: {e}");
             break;       
       }
    }
    ... 
}
```

---

however, if in your testing you find this error getting printed for normal shutdown, as Cal suggested might be possible, let's leave it as it was with the `while let`

---

Implemented the case/match statement, and currently not seeing any I/O error logs.